### PR TITLE
waylyrics: unstable-2023-05-14 -> 0.2.4

### PR DESCRIPTION
**Pull request by Aleksanaa**
## Description of changes

$out structure:

```
./result
├── bin
│   ├── waylyrics
│   └── .waylyrics-wrapped
└── share
    ├── applications
    │   └── io.poly000.waylyrics.desktop
    ├── gsettings-schemas
    │   └── waylyrics-0.2.1
    │       └── glib-2.0
    │           └── schemas
    │               ├── gschemas.compiled
    │               └── io.poly000.waylyrics.gschema.xml
    ├── icons
    │   └── hicolor
    │       └── scalable
    │           └── apps
    │               └── io.poly000.waylyrics.svg
    └── waylyrics
        └── themes
            ├── default.css
            └── default-dark.css

```

Screenshot:

2024-02-09T19:50:31,520070370+08:00

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/audio/waylyrics/default.nix [hunk 1]
```diff
@@ -1,59 +1,57 @@
...
+    cp ${./Cargo.lock} Cargo.lock
+  '';
 
-  doCheck = false; # No tests defined in the project.
+  nativeBuildInputs = [ pkg-config wrapGAppsHook4 ];
+  buildInputs = [ openssl dbus ];
 
-  WAYLYRICS_DEFAULT_CONFIG = "${placeholder "out"}/share/waylyrics/config.toml";
-  WAYLYRICS_THEME_PRESETS_DIR = "${placeholder "out"}/share/waylyrics/themes";
+  doCheck = false; # Requires Internet access
```
Comment (AndersonTorres) [inline pkgs/applications/audio/waylyrics/default.nix:36]:
```suggestion
  doCheck = false; # requires network access
```

File: pkgs/applications/audio/waylyrics/default.nix [hunk 2]
```diff
@@ -1,59 +1,57 @@
...
-      exec = "waylyrics";
-      comment = "Simple on screen lyrics for MPRIS-friendly players";
-      type = "Application";
-      icon = "io.poly000.waylyrics";
-      desktopName = "Waylyrics";
-      terminal = false;
-      categories = [ "Audio" "AudioVideo" ];
-    })
-  ];
+  env.WAYLYRICS_THEME_PRESETS_DIR = "${placeholder "out"}/share/waylyrics/themes";
```
Comment (AndersonTorres) [inline pkgs/applications/audio/waylyrics/default.nix:38]:
```suggestion
```

Use the value directly.

File: pkgs/applications/audio/waylyrics/default.nix [hunk 3]
```diff
@@ -1,59 +1,57 @@
...
 
   meta = with lib; {
-    description = "On screen lyrics for Wayland with NetEase Music source";
+    description = "On screen lyrics for Wayland with QQ and NetEase Music source";
     homepage = "https://github.com/poly000/waylyrics";
-    license = licenses.mit;
-    maintainers = [ maintainers.shadowrz ];
+    license = with licenses; [ mit cc-by-40 ];
+    maintainers = with maintainers; [ shadowrz aleksana ];
     platforms = platforms.linux;
```
Comment (AndersonTorres) [inline pkgs/applications/audio/waylyrics/default.nix:55]: Where are the Wayland dependencies?

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/applications/audio/waylyrics/default.nix (hunk 2 above)
Comment (Aleksanaa) [reply pkgs/applications/audio/waylyrics/default.nix]: Is there a good reason not to add `env`?

**Comment by teto** [comment]
weirdly enough I can't select a lyric, I click "set as lyric" but nothing happens ? 
```
2024-02-11T14:29:18.344550Z ERROR waylyrics::sync::interop::mpris::sync_task: kind="cannot get playback position"
2024-02-11T14:29:20.345906Z ERROR waylyrics::sync::interop::mpris::sync_task: kind="cannot get playback position"
2024-02-11T14:29:20.669920Z  INFO waylyrics::sync::search_window: selected 28001492 from 网易云音乐
2024-02-11T14:29:22.344335Z ERROR waylyrics::sync::interop::mpris::sync_task: kind="cannot get playback position"
```

**Comment by Aleksanaa** [comment]
Its lyrics are grabbed from NetEase Cloud (and QQ) Music. If your song does not correspond there, it will not work. The author recommends using the NetEase Cloud Music client. I used go-musicfox for the test above, but you need to log in to your account to use it.

**Comment by poly000** [comment]
> 2024-02-11T14:29:22.344335Z ERROR waylyrics::sync::interop::mpris::sync_task: kind="cannot get playback position"

seems you're using a player without MPRIS2 `Position()` support

If things are going the right way, you should see the Unsupported Player tip on lyric labels

**Comment by poly000** [comment]
@Aleksanaa could you update to v0.2.2 or above?
In v0.2.2 waylyrics fixed the issue that `QQMusic::init` panics unexpectedly
https://github.com/waylyrics/waylyrics/commit/728c11e03eeaa9fcfd5b086a57fd42fd11b4f5ff#diff-39bf1517cd24ee4ebc93a606f383a0f63316da1f050253244723e3b7d7753a2bR29

**Comment by Aleksanaa** [comment]
Done.

**Comment by Aleksanaa** [comment]
The by-name check is unrelated.

**Pull request merged**